modules/nixos/nixos.nix

```diff
@@ -0,0 +1,18 @@
+{ pkgs, config, lib, ... }:
+
+with config.lib.stylix.colors;
+
+{
+  options.stylix.targets.nixos.enable =
+    config.lib.stylix.mkEnableTarget "the NixOS logo" true;
+
+  config.nixpkgs.overlays = lib.mkIf config.stylix.targets.nixos.enable [(self: super: {
+    nixos-icons = super.nixos-icons.overrideAttrs (oldAttrs: {
+      patchPhase = ''
+        substituteInPlace logo/white.svg --replace-fail '#ffffff' '#${base05}'
+        sed -i '2i<!-- The original NixOS logo from ${oldAttrs.src.url} modified to use colors from ${scheme}. This file is licensed under https://creativecommons.org/licenses/by/4.0 -->' logo/white.svg
```

Comment:
Improve readability by wrapping:

```suggestion
        sed \
          --in-place \
          '2i<!-- The original NixOS logo from ${oldAttrs.src.url} modified to use colors from ${scheme}. This file is licensed under https://creativecommons.org/licenses/by/4.0 -->' \
          logo/white.svg
```

We should explain why the comment cannot be added to the first line:

```suggestion
        # Add the XML comment to the second line to avoid conflicts with the XML
        # declaration on the first line.
        sed \
          --in-place \
          '2i<!-- The original NixOS logo (${oldAttrs.src.url}) is licensed under https://creativecommons.org/licenses/by/4.0 and has been modified to match the '${scheme}' color scheme. -->' \
          logo/white.svg
```

I would rephrase the comment as follows:

```suggestion
        # Add the XML comment to the second line to avoid conflicts with the XML
        # declaration on the first line.
        sed \
          --in-place \
          '2i<!-- The original NixOS logo (${oldAttrs.src.url}) is licensed under https://creativecommons.org/licenses/by/4.0 and has been modified to match the '${scheme}' color scheme. -->' \
          logo/white.svg
```

Reply:
Note that the URL points to the repository as a whole so we should still include the word "from" (or something to that effect)

Reply:
> Note that the URL points to the repository as a whole so we should still include the word "from" (or something to that effect)

Agreed.

Howerver, if `logo/white.svg` stems from `oldAttrs.src.url`, then we could specify the exact file path as `<GITHUB_REPO>/tree/<REV>/logo/white.svg`. The hard-coded file URL should remain correct as long as the commands relying on `logo/white.svg` are still working.

Reply:
`oldAttrs.src` only tells us the Git URL `https://github.com/NixOS/nixos-artwork.git` or the download URL `https://github.com/NixOS/nixos-artwork/archive/488c22aad523c709c44169d3e88d34b4691c20dc.tar.gz`. By trying to construct a more specific URL, we would be relying on the source being hosted on GitHub and not some other provider.